src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -378,6 +379,8 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 					   int shardCount, bool shardCountIsStrict,
 					   char *colocateWithTableName, bool viaDeprecatedAPI)
 {
+	ErrorIfRelationIsAKnownShard(relationId);
```

Comment:
Nice to catch both distributed and reference tables with a single check!

Reply:
Could you also please add a single comment here?

Reply:
Can we maybe move this check to `EnsureCitusTableCanBeCreated`? That seems like a place we do these kinds of checks. This probably goes as the last check there